pkg/specgen/generate/container.go

```diff
@@ -33,7 +34,42 @@ func CompleteSpec(ctx context.Context, r *libpod.Runtime, s *specgen.SpecGenerat
 
 		_, mediaType, err := newImage.Manifest(ctx)
```

Comment:
Do you have to set this mediaType somewhere?

Reply:
mediaType will be used to check if its a manifest.DockerV2Schema2MediaType
https://github.com/containers/podman/pull/7798/files#diff-5a91791543058d94a51dab8d8294ca7cR75

Reply:
But if the newImage.Manifest(ctx) check fails, the nediaType is never set.

Reply:
Ah, good catch.  If we do find a suitable image and update `newImage` to point to it, we need to make sure that `mediaType` is updated to the value that correctly describes its manifest.

Reply:
If newImage.Manifest(ctx) fails, use the mediaType of the local image.